HW1/P6/P6.py

```diff
@@ -0,0 +1,95 @@
+import numpy
+import random
+import findspark
+import re
+findspark.init()
+import pyspark
+sc = pyspark.SparkContext(appName="P4")
+
+f = open("pg100.txt")
+raw_text = f.read()
+output = raw_text.split()
+
+# Clean the words in accordance with the spec
+def cleanUp(word):
+    is_word = False
+    try:
+        val = int(word)
+    except ValueError:
+        is_word = True
+    if is_word == False:
+        return False
+    if word.isupper():
+        return False
+    if word[-1] == ".":
+        if word[::-1].isupper():
+            return False
+    return True
+
+# Basic tests on cleanUp
+assert(cleanUp("lasjflks") == True)
+assert(cleanUp("JonAh") == True)
+assert(cleanUp("bllahsk28askldH") == True)
+assert(cleanUp("alksdf.") == True)
+assert(cleanUp("1234") == False)
+assert(cleanUp("51235151") == False)
+assert(cleanUp("AJSDFJKAK") == False)
+assert(cleanUp("AJSDFJKAK.") == False)
+assert(cleanUp("AJSDFAKSFIWEIEIEIEJKAK.") == False)
+
+# clean the data
+cleaned = filter(cleanUp, output)
+
+trigrams = []
+for i in range(len(cleaned) - 2):
```

Comment:
Interesting, though this probably should have been parallelized. I would recommend looking into `zipWithIndex()` with offsets to create these trigrams.